misk/src/test/kotlin/misk/web/ContentBasedDispatchTest.kt

```diff
@@ -0,0 +1,133 @@
+                .isEqualTo("json/json my friend")
+    }
+
+    @Test
+    fun postJsonExpectText() {
+        val requestContent = packetJsonAdapter.toJson(Packet("my friend"))
+        val responseContent = post(jsonMediaType, requestContent, plainTextMediaType).source()
+        assertThat(responseContent.readUtf8()).isEqualTo("json/text my friend")
+    }
+
+    @Test
+    fun postTextExpectJson() {
+        val responseContent = post(plainTextMediaType, "my friend", jsonMediaType).source()
+        assertThat(packetJsonAdapter.fromJson(responseContent)!!.message)
+                .isEqualTo("text/json my friend")
+    }
+
+    @Test
+    fun postArbitraryExpectJson() {
+        val responseContent = post(weirdTextMediaType, "my friend", jsonMediaType).source()
+        assertThat(packetJsonAdapter.fromJson(responseContent)!!.message)
+                .isEqualTo("*/json my friend")
+    }
+
+    @Test
+    fun postArbitraryExpectArbitrary() {
+        val responseContent = post(weirdTextMediaType, "my friend", weirdTextMediaType).source()
+        assertThat(responseContent.readUtf8()).isEqualTo("*/* my friend")
+    }
+
+    class TestModule : KAbstractModule() {
+        override fun configure() {
+            install(WebActionModule.create<PostJsonReturnJson>())
+            install(WebActionModule.create<PostTextReturnJson>())
+            install(WebActionModule.create<PostJsonReturnText>())
+            install(WebActionModule.create<PostAnythingReturnJson>())
+            install(WebActionModule.create<PostAnythingReturnAnything>())
+        }
+    }
+
+    class PostJsonReturnJson : WebAction {
+        @Post("/hello")
+        @RequestContentType("application/json")
+        @ResponseContentType("application/json")
+        fun hello(@misk.web.RequestBody request: Packet) = Packet("json/json ${request.message}")
+    }
+
+    class PostTextReturnJson : WebAction {
+        @Post("/hello")
+        @RequestContentType("text/plain")
+        @ResponseContentType("application/json")
+        fun hello(@misk.web.RequestBody message: String) = Packet("text/json $message")
+    }
+
+    class PostJsonReturnText : WebAction {
+        @Post("/hello")
+        @RequestContentType("application/json")
+        @ResponseContentType("text/plain")
+        fun hello(@misk.web.RequestBody request: Packet) = "json/text ${request.message}"
+    }
+
+    class PostAnythingReturnJson : WebAction {
+        @Post("/hello")
+        @ResponseContentType("application/json")
+        fun hello(@misk.web.RequestBody message: String) = Packet("*/json $message")
+    }
+
+    class PostAnythingReturnAnything : WebAction {
+        @Post("/hello")
+        fun hello(@misk.web.RequestBody message: String) = "*/* $message"
+    }
+
+    private fun post(contentType: MediaType, content: String,
+            acceptedMediaType: MediaType? = null): okhttp3.ResponseBody {
+        val httpClient = OkHttpClient()
```

Comment:
we gotta somehow DI these. Each call to this constructor is slow ’cause it creates its own threadpools